pkg/report/github_sbom.go

```diff
@@ -0,0 +1,186 @@
+package report
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	ftypes "github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/purl"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+const (
+	DirectRelationship   string = "direct"
+	IndirectRelationship string = "indirect"
+	RuntimeScope         string = "runtime"
+	DevelopmentScope     string = "development"
+)
+
+type GithubSbomPackage struct {
+	PackageUrl   string   `json:"package_url,omitempty"`
+	Relationship string   `json:"relationship,omitempty"`
+	Dependencies []string `json:"dependencies,omitempty"`
+	Scope        string   `json:"scope,omitempty"`
+	Metadata     Metadata `json:"metadata,omitempty"`
+}
+
+type GithubSbomFile struct {
+	SrcLocation string `json:"source_location,omitempty"`
+}
+
+type Metadata map[string]interface{}
+
+type GithubSbomManifest struct {
+	Name     string                       `json:"name,omitempty"`
+	File     *GithubSbomFile              `json:"file,omitempty"`
+	Metadata Metadata                     `json:"metadata,omitempty"`
+	Resolved map[string]GithubSbomPackage `json:"resolved,omitempty"`
+}
+
+type GithubSbomJob struct {
+	Correlator string `json:"correlator,omitempty"`
+	Id         string `json:"id,omitempty"`
+}
+type GithubSbomDetector struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+	Url     string `json:"url"`
+}
+
+type GithubSbom struct {
+	Version   int                           `json:"version,omitempty"`
+	Detector  GithubSbomDetector            `json:"detector"`
+	Metadata  Metadata                      `json:"metadata,omitempty"`
+	Ref       string                        `json:"ref,omitempty"`
+	Sha       string                        `json:"sha,omitempty"`
+	Job       *GithubSbomJob                `json:"job,omitempty"`
+	Scanned   string                        `json:"scanned,omitempty"`
+	Manifests map[string]GithubSbomManifest `json:"manifests,omitempty"`
+}
+
+type GithubSbomWriter struct {
+	Output  io.Writer
+	Version string
+}
+
+func init() {
+	CustomTemplateFuncMap["now"] = time.Now
+	CustomTemplateFuncMap["getenv"] = os.Getenv
+}
+
+func (gsbmw GithubSbomWriter) Write(report types.Report) error {
+	getenv, ok := CustomTemplateFuncMap["getenv"].(func(string) string)
+	if !ok {
+		return xerrors.Errorf("invalid getenv reference")
+	}
+	githubSbom := &GithubSbom{}
+
+	//use now() method that can be overwritten while integration tests run
+	githubSbom.Scanned = CustomTemplateFuncMap["now"].(func() time.Time)().Format(time.RFC3339)
+	githubSbom.Detector = GithubSbomDetector{
+		Name:    "trivy",
+		Version: gsbmw.Version,
+		Url:     "https://github.com/aquasecurity/trivy",
+	}
+	githubSbom.Version = 0 // The version of the repository snapshot submission. It's not clear what value to set
```

Comment:
@ShiraCohen33 
Thank you for clarifying required version, but could you also change comment?